### PR TITLE
Fixes #11913: no reporting if value to report contains a variable

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://www.rudder-project.org/redmine/issues/11913

### CODE REVIEW

File: tools/ncf.py [hunk 1]
```diff
@@ -547,7 +547,11 @@ def generate_technique_content(technique, methods):
     else:
       promiser = "method_call"
 
-    content.append('    "'+promiser+'" usebundle => '+method_call['method_name']+'('+arg_value+'),')
+    class_parameter_id    = method_info["class_parameter_id"]
+    class_parameter_name  = method_info["parameter"][class_parameter_id]["name"]
+    class_parameter_value = method_call["args"][class_parameter_id]
+    content.append('    "'+promiser+'_context" usebundle => current_technique_report_info("'+method_info["name"]+'", "'+class_parameter_name+'", "'+class_parameter_value+'");')
```
Comment (ncharles) [inline tools/ncf.py]: is it properly escaped ? what if it component key contains " or ' ?

Comment (peckpeck) [reply tools/ncf.py]: This doesn't match current_technique_report_info call, the first parameter is technique_name

Comment (VinceMacBuche) [reply tools/ncf.py]: Thanks @ncharles ! this needs escaping you are right :( :(

Comment (VinceMacBuche) [reply tools/ncf.py]: Thanks @peckpeck  it should be the technique name not the method name ...

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
PR rebased

**Comment by VinceMacBuche** [comment]
Commit modified

File: tools/ncf.py [hunk 2]
```diff
@@ -550,6 +550,7 @@ def generate_technique_content(technique, methods):
   for method_call in technique["method_calls"]:
     method_name = method_call["method_name"]
     method_info = methods[method_name]
+    # regex to match quote characters not preceded by a backslash
```
Comment (ncharles) [inline tools/ncf.py]: it seems to me it does a little more than backslash, or else the regexes lines 581-583 are not complete

File: tree/20_cfe_basics/log_rudder.cf [hunk 3]
```diff
@@ -29,212 +29,17 @@
...
-      # 5/ we aggregate this with the availability of the promiser stack to know if we should report using v2 format
-      "report_v2"           expression => "expected_report_v2_csv.expected_report_v2_res.method_v2.has_promiser_stack.class_prefix_v2";
-
-    any::
-      "no_old_class_prefix" expression => strcmp("${old_class_prefix}", "");
-
-      "pass3" expression => "pass2";
-      "pass2" expression => "pass1";
-      "pass1" expression => "any";
+    "c_class_prefix" string => canonify("${old_class_prefix}");
```
Comment (ncharles) [inline tree/20_cfe_basics/log_rudder.cf]: could you add 2 spaces of indentation ?

Comment (VinceMacBuche) [reply tree/20_cfe_basics/log_rudder.cf]: It's correctly idented, there is no class defined

<pre>
  vars:
    "c_class_prefix" ...
</pre>

File: tools/ncf.py (hunk 2 above)
Comment (VinceMacBuche) [reply tools/ncf.py]: No it's exactly that, the regex here in 554 is using a negative lookbehind: 

> (?<!text)

to match quotes not preceded by a backslash

then 581-583 use the regexp defined in 554 to replace the unescaped quotes by an escaped quote

File: tree/20_cfe_basics/log_rudder.cf (hunk 3 above)
Comment (VinceMacBuche) [reply tree/20_cfe_basics/log_rudder.cf]: or do indented - spacedces even iff there is no class ?

Comment (ncharles) [reply tree/20_cfe_basics/log_rudder.cf]: yes, even if no class

Comment (VinceMacBuche) [reply tree/20_cfe_basics/log_rudder.cf]: OK from what i read elsewhere we do likewise !! thanks @ncharles  !

File: tools/ncf.py (hunk 2 above)
Comment (ncharles) [reply tools/ncf.py]: My bad ! Thank you for the clarification

**Comment by VinceMacBuche** [comment]
Commit modified

File: tree/20_cfe_basics/log_rudder.cf (hunk 3 above)
Comment (VinceMacBuche) [reply tree/20_cfe_basics/log_rudder.cf]: ok i reindented with 6 spaced (and the report below is also now indented with 6 spaces)

**Pull request merged**